test/test_device.cpp

```
@@ -0,0 +1,829 @@
+  const auto result = AWS_QDMI_device_session_query_device_property(
+      session, QDMI_DEVICE_PROPERTY_DURATIONSCALEFACTOR, sizeof(double),
+      &scaleFactor, nullptr);
+  EXPECT_THAT(result, testing::AnyOf(QDMI_SUCCESS, QDMI_ERROR_NOTSUPPORTED));
+  if (result == QDMI_SUCCESS) {
+    EXPECT_GT(scaleFactor, 0.);
+  }
+}
+
+TEST_F(AWSQDMISpecificationTest, QuerySiteIndex) {
+  size_t id = 0;
+  EXPECT_NO_THROW(for (auto* site : querySites(session)) {
```

Cppcheck (2.19.0)

[error] There is an unknown macro here somewhere. Configuration is required. If EXPECT_NO_THROW is a macro then please configure it.

(unknownMacro)